tests/system-tests/internal/supporttools/tcpdump.go

```diff
@@ -0,0 +1,543 @@
+package supporttools
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift-kni/eco-gotests/tests/system-tests/internal/apiobjectshelper"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/deployment"
+	"github.com/openshift-kni/eco-goinfra/pkg/pod"
+	scc "github.com/openshift-kni/eco-gotests/tests/system-tests/internal/scc"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	httpProtocol     = "http"
+	udpProtocol      = "udp"
+	tcpCaptureScript = `#!/bin/bash
+tcpdump -vvv -e -nn -i %s | egrep -i 'Host:'
+`
+
+	udpCaptureScript = `#!/bin/bash
+tcpdump -n -s0  -i %s port %s and udp
+`
+	logScanMaxTries = 3
+)
+
+// CreateTCPDumpDeployment creates support-tools tcpdump deployment on the nodes specified in scheduleOnNodes.
+func CreateTCPDumpDeployment(
+	apiClient *clients.Settings,
+	tdNamespace,
+	tdDeploymentLabel,
+	tdImage,
+	packetCaptureInterface,
+	packetCaptureProtocol string,
+	scheduleOnNodes []string) (*deployment.Builder, error) {
+	if packetCaptureProtocol != httpProtocol && packetCaptureProtocol != udpProtocol {
+		return nil, fmt.Errorf("CreateTraceRouteDeployment supports only %s and "+
+			"%s protocols, got: %s", httpProtocol, udpProtocol, packetCaptureProtocol)
+	}
+
+	glog.V(100).Infof("Getting the image and tag for tools from the release image")
+
+	stDeploymentName := "tcpdump-tool"
+
+	glog.V(100).Infof("Create support-tools namespace %s", tdNamespace)
+
+	err := ensureNamespaceExists(apiClient, tdNamespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create support-tools namespace %s: %w", tdNamespace, err)
+	}
+
+	glog.V(100).Infof("Adding SCC privileged to the support-tools namespace")
+
+	err = scc.AddPrivilegedSCCtoDefaultSA(tdNamespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to add SCC privileged to the supporttools namespace %s: %w",
+			tdNamespace, err)
+	}
+
+	stDeployment, err := createTCPDumpDeployment(
+		apiClient,
+		tdImage,
+		stDeploymentName,
+		tdNamespace,
+		tdDeploymentLabel,
+		packetCaptureInterface,
+		packetCaptureProtocol,
+		scheduleOnNodes,
+	)
+
+	if err != nil {
+		return stDeployment, err
+	}
+
+	retries := 3
+	pollInterval := 5
+
+	for i := 0; i < retries; i++ {
```

Comment:
won't it be easier to use `deployment.WaitUntilCondition("Available", 1*time.Minute)`  rather than implement own retry loop?

Reply:
make sense. but in this case await.WaitUntilAllDeploymentsReady will suite better, wdyt?